tools/simulator/polySimulator.cpp

```diff
@@ -0,0 +1,801 @@
+//
```

Comment:
I see a lot of code duplicate from the simulator class. Please migrate it to the simulator class.
You always can distinguish between your or the regular simulator unique features by variable from general settings